ext/sockets/sockets.c

```diff
@@ -1580,7 +1580,8 @@ PHP_FUNCTION(socket_recvfrom)
 
 			if (arg6 == NULL) {
 				zend_string_efree(recv_buf);
-				WRONG_PARAM_COUNT;
+				zend_argument_value_error(6, "must be specified when argument #1 ($socket) is of type AF_INET or AF_INET6");
```

Comment:
I believe this should stay ArgumentCountError, especially since it's an out parameter.

How about:

> socket_recvfrom() expects exactly 6 arguments when argument #1 ($socket) is of type AF_INET or AF_INET6, $x given.